File: Learning/OtherLanguages/C/C练习题/Use C/17.c
#include <stdio.h>
#include <stdlib.h>

void main()
{
    char ch;
    int letter, space, number, other;
    letter = space = number = other = 0;

    while((ch = getchar()) != '\n')
    {
        if((ch>='A'&&ch<='Z')||(ch>='a'&&ch<='z'))
        {
            letter++;
        }
        else if(ch>='0' && ch<='9')
        {
            number++;
        }
        else if(' ' == ch)
        {
            space++;
        }
        else
        {
            other++;
        }
    }
    printf("There are %d letters, %d numbers, %d spaces, %d others\n", letter, number, space, other);
}

File: Learning/OtherLanguages/C/C练习题/Use Python/17.py
string = input('请输入：')


letter, digit, space, other = 0, 0, 0, 0
for each in string:
    if (each >= 'a' and each <='z') or (each >= 'A' and each <= 'Z'):
        letter += 1
    elif each >= '0' and each <= '9':
        digit += 1
    elif each == ' ':
        space += 1
    else:
        other += 1
        
print('有英文字母%d个\n有数字%d个\n有空格%d个\n有其他字符%d个' \
      % (letter, digit, space, other))
